### ISSUE TITLE
Try threaded unit_test_runner.ts

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno#4792]
With `Deno` API now available in `Worker` (#4784) we could try dogfooding it a bit by creating yet another version of `cli/js/tests/unit_test_runner.ts` that would use worker threads instead of subprocesses.

Ideas: 
a) in `master` thread load and compile all files listed in `cli/js/tests/unit_tests.ts` (that's already done)

b) build mapping with required permissions combination for each file, ie. `PermissionSetTestResult` should be extended to keep list of files using given permission combination:
```ts
{
    perms: {
        read: true,
        write: true,
    },
    files: [
        "cli/js/tests/file_test.ts",
        ....
    ]
}
```

c) add small `unit_test_worker.ts` runtime that is used to create test workers. It should expect only a single message describing which permissions to use and which file to load.

d) for each permissions combination; and for each file in that combination (described in point b) ) create new worker using runtime from point c) and run all tests matching permissions. Report results back to `master` thread

e) Multiple workers should be running in parallel - probably as many as there are cores on machines, though I'm not sure if we expose a method to get CPU count. First pass can have hard limit of 8 parallel workers.
